File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPAllowFlight.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPAllowFlight implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "allow_flight";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return false;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.getAllowFlight();
    }

    @Override
    public void restoreProperty(Player player, Boolean allow) {
        player.setAllowFlight(allow);
    }

    @Override
    public Class<Boolean> getTypeClass() {
        return Boolean.class;
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPAllowFlight.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPAllowFlight extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "allow_flight"

  override def getDefaultSerializedProperty: Boolean = false

  override def getSerializedPropertyFromPlayer(player: Player): Boolean = player.getAllowFlight

  override def restoreProperty(player: Player, allow: Boolean): Unit = player.setAllowFlight(allow)
